Azure/sonic-buildimage
# [asan] Add address sanitizer support.

Implement infrastructure that allows enabling address sanitizer
for docker containers. Enable address sanitizer for SWSS container.



#### Why I did it
To add a possibility to compile SONiC applications with address sanitizer (ASAN). 
ASAN is a memory error detector for C/C++. It finds:
- Use after free (dangling pointer dereference)
- Heap buffer overflow
- Stack buffer overflow
- Global buffer overflow
- Use after return
- Use after the scope
- Initialization order bugs
- Memory leaks

#### How I did it
By adding new **ENABLE_ASAN** configuration option.
#### How to verify it
By default **ASAN** is disabled and the SONiC image is not affected.
When **ASAN** is enabled it inspects all allocation, deallocation, and memory usage that the application does in run time. To verify whether the application has memory errors tests that trigger memory usage of the application should be run. Ideally, the whole regression tests should be run. Memory leaks reports will be placed in **/var/log/asan/** directory of SONiC host OS.
#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog



#### A picture of a cute animal (not mandatory but encouraged)
